topiary-core/src/lib.rs

```diff
@@ -260,16 +259,18 @@ pub fn formatter(
             log::info!("Pretty-print output");
             let rendered = pretty::render(
                 &atoms[..],
-                // Default to "  " is the language has no indentation specified
+                // Default to "  " if the language has no indentation specified
                 language.indent.as_ref().map_or("  ", |v| v.as_str()),
             )?;
-            let trimmed = trim_whitespace(&rendered);
+
+            // Add a final line break if missing
+            let rendered = format!("{}\n", rendered.trim());
```

Comment:
This is a niche edge case, but if the input has no EOF `\n` and contains semantic horizontal whitespace at the end of the last line, that will get trimmed.